src/Sprites/IndexedBmpWriter.cpp

```diff
@@ -130,19 +136,20 @@ void IndexedBmpWriter::VerifyPixelBufferSizeMatchesImageDimensions(uint16_t bitC
 	}
 }
 
-void IndexedBmpWriter::VerifyPixelBufferSizeMatchesImageDimensionsWithPitch(uint16_t bitCount, int32_t width, int32_t height, std::size_t pixelCountIncludingPitch)
+void IndexedBmpWriter::VerifyPixelBufferSizeMatchesImageDimensionsWithPitch(uint16_t bitCount, int32_t width, int32_t height, std::size_t pixelsWithPitchSize)
 {
-	const uint16_t pixelsPerByte = 8 / bitCount;
-
-	if (pixelCountIncludingPitch * pixelsPerByte != CalcScanlinePitch(bitCount, width) * std::abs(height)) {
+	// G++ will flag warning -Wsign-compare if comparing a signed and unsigned value
+	if (pixelsWithPitchSize != static_cast<uint32_t>(CalculatePitchSize(bitCount, width) * std::abs(height))) {
 		throw std::runtime_error("An incorrect number of pixels were passed.");
 	}
 }
 
 void IndexedBmpWriter::VerifyPixelsContainedInPalette(uint16_t bitCount, std::size_t paletteEntryCount, const std::vector<uint8_t>& pixels)
 {
 	// Check if palette is full
-	if (paletteEntryCount == 1 << bitCount) {
+	// G++ will flag warning -Wsign-compare if comparing a signed and unsigned value
+	// MSVC will flag warning C4334 on x64 compilation if bit shift not set to 64 bit integer
+	if (paletteEntryCount == static_cast<uint64_t>(1) << bitCount) {
```

Comment:
Using a suffix of `u` here would make `1u` an unsigned value. That eliminates the warning on g++ and clang. Alternatively a suffix of `ull` can also be used to be explicit about the size, which also eliminates warnings on both g++ and clang. I hadn't seen the `ui64` suffix before. I suspect it's less common, or part of a newer compiler standard.

My recommendation here is to use the `u` suffix. In particular, we are dealing with a `std::size_t` value on the left, so the size may vary with compiler settings.

We should prefer suffixes over casts. Using a cast defeats many compiler checks, and so is more prone to bugs.